cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -90,6 +90,7 @@ template <typename Rep,
                                            is_supported_representation_type<Rep>())>* = nullptr>
 CUDA_HOST_DEVICE_CALLABLE Rep ipow(T exponent)
 {
+  assert(("integer exponentiation with negative exponent is not possible.", exponent >= 0));
```

Comment:
Just checking: Recently someone told us we should not use the comma operator in libcudf...  This is using the comma operator, right?

Also, why is this only an assert, shouldn't it be a `CUDF_EXPECTS`?

Reply:
`CUDF_EXPECT` don't work in device code. This is the idiomatic way to get a "error message" using an `assert`. If you search for `assert` in this file, you will see a couple other places.

Reply:
This is what `release_assert` is for https://github.com/rapidsai/cudf/blob/2780a8cea7743002d62b72c804ef693c8807d586/cpp/include/cudf/detail/utilities/release_assert.cuh#L21-L37

Reply:
Fixed!